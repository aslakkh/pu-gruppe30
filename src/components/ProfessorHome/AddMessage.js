/**
 * Created by jan on 21/03/2017.
 */
import React, { Component } from 'react'
import firebase from 'firebase';
import {ref } from '../../config/constants'
import {Form, FieldGroup, FormControl} from 'react-bootstrap'


/*
* Lets the professor add a message the student can see
 */

export default class AddMessage extends Component {
    constructor(props){
        super(props);
        this.state={
            courseID:this.props.courseID,
        }
    }
    handleSubmit = (e) =>{
        e.preventDefault();
        this.castMessageToFirebase()

    }
    /*
    * Send the message to firebase and uses timestap as ID
    */
     castMessageToFirebase(){
         const userUid = firebase.auth().currentUser.uid;
         console.log(this.state.courseID)
         console.log(this.message.value)
         console.log('courses/'+this.state.courseID+'/Messages/'+ Date.now())
         const timeRef = ref.child('courses/'+this.state.courseID+'/Messages/'+ Date.now());
         timeRef.set({
             Message: this.message.value
         })
         this.message.value=""


    }


    render(){
        return(


            <form onSubmit={this.handleSubmit}>
                <FormControl
                    id="formControlsText"
                    type="text"
                    inputRef={ (ref) => {this.message = ref;}}
                    label="Message"
                    placeholder="Message"
                />
                <button type="submit" className="btn btn-primary">Submit</button>

            </form>

        )
    }


}