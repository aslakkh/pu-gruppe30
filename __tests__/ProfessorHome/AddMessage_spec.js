/**
 * Created by jan on 21/04/2017.
 */
/**
 * Created by jan on 19/04/2017.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import AddMessage from '../../src/components/ProfessorHome/AddMessage.js';
import TestUtils from "react-addons-test-utils"
import {mount} from 'enzyme'
import jest from 'jest'


describe("Add Message", function () {
    let renderer2;
    let component2;
    renderer2 = TestUtils.createRenderer();
    renderer2.render(
        <AddMessage courseID={"TES1000"}/>
    );
    component2 = renderer2.getRenderOutput();
    it("Not logged in", () => {
        expect(component2).toMatchSnapshot();
    });




    it('Test start button', () => {

        const wrapper = mount(<AddMessage courseID={"TES1000"}/>);
        const input = wrapper.find('input')
        input.simulate('change', { target: { value: 'Hello' } });
        wrapper.find('button').simulate('click')
        expect(wrapper.state('value')).toBe('Hello');
       //wrapper.find('input').get(0).simulate('change', {target: {value: 'My new value'}});

    })

});
/*
 test('renders home if not logged in', () => {
 const component = renderer.create(
 <Home />
 ).toJSON();
 expect(component).toMatchSnapshot(); //renders h4 if no courses in props
 const component2 = renderer.create(
 <Home courses={courses} authed={true} admin={true}/>
 ).toJSON();
 expect(component2).toMatchSnapshot();
 })*/