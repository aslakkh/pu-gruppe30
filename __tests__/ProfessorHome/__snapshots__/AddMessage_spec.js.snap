// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Message Not logged in 1`] = `
<div>
  <h3>
    TES1000
  </h3>
  <form
    onSubmit={[Function]}
  >
    <FormControl
      bsClass="form-control"
      componentClass="input"
      id="formControlsText"
      inputRef={[Function]}
      label="Message"
      onChange={[Function]}
      placeholder="Message"
      type="text"
      value=""
    />
    <button
      className="btn btn-primary"
      disabled={true}
      style={
        Object {
          "marginTop": "10px",
        }
      }
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
